pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -255,19 +298,9 @@ func (admitter *VMsAdmitter) applyInstancetypeToVm(vm *v1.VirtualMachine) (*inst
 		}}
 	}
 
-	if topology := instancetype.GetPreferredTopology(preferenceSpec); instancetypeSpec != nil && topology == instancetypev1beta1.PreferSpread {
-		ratio := preferenceSpec.PreferSpreadSocketToCoreRatio
-		if ratio == 0 {
-			ratio = instancetype.DefaultSpreadRatio
-		}
-
-		if (instancetypeSpec.CPU.Guest % ratio) > 0 {
-			return nil, nil, []metav1.StatusCause{{
-				Type:    metav1.CauseTypeFieldValueInvalid,
-				Message: "Instancetype CPU Guest is not divisible by PreferSpreadSocketToCoreRatio",
-				Field:   k8sfield.NewPath("instancetype.spec.cpu.guest").String(),
-			}}
-		}
+	spreadvCPUConflicts := checkSpreadCPUTopology(instancetypeSpec, preferenceSpec)
+	if len(spreadvCPUConflicts) > 0 {
```

Comment:
Inline assignment into if?

Reply:
Yup can do, the line above is already a little long but this isn't python ;)

Reply:
Up to you, but to me it makes sense. You could also shorten the variable's name if you'd like. :)